File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*args)
    @args = args.sort
  end

  def to(limit)
    @args.map do |mult|
      if limit < mult
        0
      else
        (mult..limit-1).to_a.map do |n|
          n if n % mult == 0
        end
      end
    end.flatten.compact.uniq.reduce(:+)
  end
end

File: elixir/sum-of-multiples/sum_of_multiples.exs
defmodule SumOfMultiples do
  def to(limit, factors) do
    factors
    |> Enum.map(fn(fact) ->
      if fact < limit do
        Enum.to_list(fact..limit-1)
        |> Enum.map(fn(v) -> if rem(v, fact) == 0, do: v, else: nil end)
      end
    end)
    |> List.flatten
    |> Enum.reject(fn(x) -> is_nil(x) == true end)
    |> Enum.uniq
    |> Enum.sum
  end
end
